controllers/signer/interface.go

```diff
@@ -49,3 +51,21 @@ type CertificateRequestObject interface {
 
 	GetConditions() []cmapi.CertificateRequestCondition
 }
+
+// IgnoreIssuer is an optional function that can prevent the issuer controllers from
+// reconciling an issuer resource. By default, the controllers will reconcile all
+// issuer resources that match the owned types.
+type IgnoreIssuer func(
+	ctx context.Context,
+	issuerObject v1alpha1.Issuer,
+) (bool, error)
+
+// IgnoreCertificateRequest is an optional function that can prevent the CertificateRequest
+// and Kubernetes CSR controllers from reconciling a CertificateRequest resource. By default,
+// the controllers will reconcile all CertificateRequest resources that match the issuerRef type.
+type IgnoreCertificateRequest func(
+	ctx context.Context,
+	cr CertificateRequestObject,
+	issuerGvk schema.GroupVersionKind,
+	issuerName types.NamespacedName,
+) (bool, error)
```

Comment:
suggestion: My understanding of these functions is that someone implementing an issuer using issuer-lib may want to define them for their issuer.

If that's correct, would it be useful in the comments for these functions to describe when and how they're called? At the moment I can understand that these functions _can_ prevent reconciliation but I don't know _how_ or _when_ they do!

Reply:
I tried to improve the comments PTAL.